src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -231,17 +231,14 @@ public HttpConnectionDispatcher(ConnectionManager manager, ILoggerFactory logger
 
         private ConnectionState CreateConnection(HttpContext context)
         {
-            var format =
-                string.Equals(context.Request.Query["format"], "binary", StringComparison.OrdinalIgnoreCase)
-                    ? MessageType.Binary
-                    : MessageType.Text;
-
             var state = _manager.CreateConnection();
             state.Connection.User = context.User;
 
             // TODO: this is wrong. + how does the user add their own metadata based on HttpContext
             var formatType = (string)context.Request.Query["formatType"];
             state.Connection.Metadata["formatType"] = string.IsNullOrEmpty(formatType) ? "json" : formatType;
+            state.Connection.Metadata["Context"] = context;
```

Comment:
Sure. I just took this: https://github.com/aspnet/SignalR/pull/390/files#diff-366b10d924ffee156b8529bd1727f6e2R236